pypeit/traceslits.py

```diff
@@ -1017,141 +1059,185 @@ def __repr__(self):
         txt += '>'
         return txt
 
-
-    def save_master(self, tslits_dict, mstrace=None, outfile=None, overwrite=True):
+    def save(self, outfile=None, overwrite=True, traceImage=None):
```

Comment:
So I previously changed the interface to these master frames to take an argument. For save this is the master to be written, for load it is the file to read from. There are important  reasons for that. 

At times I need to be able to write master files to disk using the save_master method without running the entire class. This is the case when I create pseudo masters as are the case for the 2d-coadds.  You could fix that by for example adding an optional argument where the master is specified instead of using the attributed. Otherwise I have to instantiate the class and set class attributes to the thing I want to write which is pretty tedious. Another solution would be for this save method to just wrap some core method that takes arguments.  

Here is an example from the 2d coadd codes: 


    # Write out the psuedo master files to disk
    master_key_dict = stack_dict['master_key_dict']
    waveImage = waveimage.WaveImage(None, None, None, None, None,
                                      master_key=master_key_dict['arc'], master_dir=master_dir)
    waveImage.save_master(waveimg_psuedo)
    traceSlits = traceslits.TraceSlits(None, None, None, master_key=master_key_dict['trace'], master_dir=master_dir)
    traceSlits.save_master(tslits_dict_psuedo)

I think you can give me this functionality back (which you've removed) by just adding optional arguments. 

The scripts often need to load these masters and that is why those utility functions exist. It looks like you changed those into static methods which I agree is a better approach. 

Anyway I think these are easy fixes. Ideal you would have caught this conflict with 2d coadds via a test, but the 2d coadds are not fully implemented yet.

Reply:
Here's the fix I've implemented:
```
waveImage = waveimage.WaveImage(None, None, None, None, None,
                                master_key=master_key_dict['arc'], master_dir=master_dir)
waveImage.save(mswave=waveimg_psuedo)

traceSlits = traceslits.TraceSlits(None, None, master_key=master_key_dict['trace'],
                                   master_dir=master_dir)
traceSlits.save(tslits_dict=tslits_dict_psuedo)
```

This should only be a stop-gap until we come up with a better solution.

Reply:
Thanks @kbwestfall, this looks great. I'm happy to discuss better solutions, but I'm afraid these pseudo masters, although not very elegant, are something that we have to live with.